powertools-large-messages/src/main/java/software/amazon/lambda/powertools/largemessages/internal/LargeSQSMessageProcessor.java

```diff
@@ -0,0 +1,160 @@
+     */
+    static Optional<String> calculateMessageBodyMd5(String messageBody) {
+        byte[] expectedMd5;
+        try {
+            expectedMd5 = Md5Utils.computeMD5Hash(messageBody.getBytes(StandardCharsets.UTF_8));
+        } catch (Exception e) {
+            LOG.warn("Unable to calculate the MD5 hash of the message body. ", e);
+            return Optional.empty();
+        }
+        return Optional.of(BinaryUtils.toHex(expectedMd5));
+    }
+
+    /**
+     * Compute the MD5 of the message attributes.<br/>
+     * Inspired from {@code software.amazon.awssdk.services.sqs.internal.MessageMD5ChecksumInterceptor}.<br/>
+     * package protected for testing purpose.
+     *
+     * @param messageAttributes attributes of the SQS Message
+     * @return the MD5 digest of the SQS Message attributes (or empty in case of error)
+     */
+    @SuppressWarnings("squid:S4790") // MD5 algorithm is used by SQS, we must use MD5
+    static Optional<String> calculateMessageAttributesMd5(final Map<String, MessageAttribute> messageAttributes) {
+        List<String> sortedAttributeNames = new ArrayList<>(messageAttributes.keySet());
+        Collections.sort(sortedAttributeNames);
+
+        MessageDigest md5Digest;
+        try {
+            md5Digest = MessageDigest.getInstance("MD5");
+
+            for (String attrName : sortedAttributeNames) {
+                MessageAttribute attrValue = messageAttributes.get(attrName);
+
+                // Encoded Name
+                updateLengthAndBytes(md5Digest, attrName);
+
+                // Encoded Type
+                updateLengthAndBytes(md5Digest, attrValue.getDataType());
+
+                // Encoded Value
+                if (attrValue.getStringValue() != null) {
+                    md5Digest.update(STRING_TYPE_FIELD_INDEX);
+                    updateLengthAndBytes(md5Digest, attrValue.getStringValue());
+                } else if (attrValue.getBinaryValue() != null) {
+                    md5Digest.update(BINARY_TYPE_FIELD_INDEX);
+                    updateLengthAndBytes(md5Digest, attrValue.getBinaryValue());
+                } else if (attrValue.getStringListValues() != null &&
+                        attrValue.getStringListValues().size() > 0) {
+                    md5Digest.update(STRING_LIST_TYPE_FIELD_INDEX);
+                    for (String strListMember : attrValue.getStringListValues()) {
+                        updateLengthAndBytes(md5Digest, strListMember);
+                    }
+                } else if (attrValue.getBinaryListValues() != null &&
+                        attrValue.getBinaryListValues().size() > 0) {
+                    md5Digest.update(BINARY_LIST_TYPE_FIELD_INDEX);
+                    for (ByteBuffer byteListMember : attrValue.getBinaryListValues()) {
+                        updateLengthAndBytes(md5Digest, byteListMember);
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Same as above - do we really want to swallow and silently null if we fail to do this?

Reply:
I didn't want to return null, so returned empty but it's the same in the end...